lib/hash.ts

```diff
@@ -0,0 +1,57 @@
+// This module implements the "double hash" signature specification described in docs/signature_specification.md
+import * as crypto from 'crypto';
+import { FileContent, FilePath, FullFileHash, SignatureResult } from './types';
+import { isBinary } from './utils/binary';
+import { removeWhitespaces } from './utils/format';
+
+enum DigestFormat {
+  BASE64 = 'base64',
+  HEX = 'hex',
+}
+
+enum HashAlgorithm {
+  MD5 = 'md5',
+  OTHER = 'other',
+}
+
+const usedHashAlgorithm: HashAlgorithm = HashAlgorithm.MD5;
+
+export async function getHashSignature(
+  path: FilePath,
+  content: FileContent,
+): Promise<SignatureResult> {
+  const hashes: FullFileHash[] = await Promise.all([
+    computeHash(content),
+    computeUHash(content),
+  ]);
+
+  return {
+    path: path,
+    size: content.length,
+    hashes_ffm: hashes,
+  };
+}
+
+export async function computeHash(content: FileContent): Promise<FullFileHash> {
+  const hash = crypto.createHash(usedHashAlgorithm).update(content);
```

Comment:
I think when the time comes and the crypto won't be able to cover our hashing algos we can create then a 'facade' to encapsulate the logic and be able to switch between different vendors. What do you say?

Reply:
We've decided to remove future-proofing to avoid possible future confusion, in favour of adding any new implementations when needed; the function's been renamed to `computeSingleHash` to reflect its specific application.